### PR TITLE
Permutation python 1

### PR DESCRIPTION
**Pull request by Kunal-Kumar-Sahoo**
### **Describe your change:**
Created permutations.py program to calculate permutations of objects

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/permutations.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"""
+Reference: https://en.wikipedia.org/wiki/Permutation
+"""
+
+from math import factorial
+
+def permutations(n: int, r: int) -> int:
```
[the remark this answers is not in the document]
Comment (Kunal-Kumar-Sahoo) [reply maths/permutations.py:7]: Can anyone help me with the doctest requirements as I took reference from combinations.py and I couldn't understand doctest requirement from that script.

File: maths/permutations.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    Returns the number of different ways of r length which can
+    be made from n values where relative order of choice matters 
+    and n >= k.
+
+    Examples:
+    >>>permutations(10, 5)
```
Comment (cclauss) [inline maths/permutations.py:14]:
```suggestion
    >>> permutations(10, 5)
```
The space is important.

File: maths/permutations.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    """
+
+    # If either of the conditions are true, the function is being asked
+    # to calculate a factorial of a negative number, which is not possible
+    if n < k or k < 0:
+        raise ValueError("Please enter positive integers for n and k where n >= k")
+    return int(factorial(n) / (factorial(n - k)))
+
+if __name__ == "__main__":
+    print(permutations(10, 5))
```
Comment (abhijitgawai) [inline maths/permutations.py:39]: I am new here. Why do we need to write this

Comment (cclauss) [reply maths/permutations.py:39]: Line 38 will be True when the user runs the program so they will see what the function returns.
Line 38 will be False when a testing tool like pytest opens the file to look for and run tests.

File: maths/permutations.py (hunk 2 above)
Comment (Kunal-Kumar-Sahoo) [reply maths/permutations.py]: I implemented the corrections locally on my system and pushed it to the fork of the project in my account. Can you help me how to update it over here?

**Comment by Kunal-Kumar-Sahoo** [comment]
@cclauss I recieved a notification that checks are failing. Can you please help me sort it out?

**Comment by cclauss** [comment]
Clicking `Details` to the right of the failing tests shows:
`E   ValueError: line 20 of the docstring for maths.permutations.permutations lacks blank after  ..: ' ...'`

File: maths/permutations.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    
+    >>> permutations(20, 1)
+    20
+
+    >>> permutations(0, 0)
+    1
+
+    >>> permutations(-4, -5)
+     ...
+    Traceback (most recent call last):
```
Comment (cclauss) [inline maths/permutations.py:28]:
```suggestion
    Traceback (most recent call last):
     ...
```

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**